src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -46,9 +46,9 @@ class SamAuthProvider(val config: Config, serviceAccountProvider: ServiceAccount
       new CacheLoader[SamCacheKey, Future[Boolean]] {
         def load(key: SamCacheKey) = {
           key match {
-            case NotebookAuthCacheKey(userInfo, action, googleProject, clusterName, executionContext) =>
+            case NotebookAuthCacheKey(internalId, userInfo, action, googleProject, clusterName, executionContext) =>
               // tokenExpiresIn should not taken into account when comparing cache keys
-              hasNotebookClusterPermissionInternal(userInfo.copy(tokenExpiresIn = 0), action, googleProject, clusterName)(executionContext)
+              hasNotebookClusterPermissionInternal(internalId, userInfo.copy(tokenExpiresIn = 0), action, googleProject, clusterName)(executionContext)
```

Comment:
wonder why the executionContext needs to be passed expliclity

Reply:
This is kind of a hack to pass the implicit EC to the guava cache loading function. Implicits aren't threaded through into guava, so the only way for the cache loader to get it is via the cache key. I tested once upon a time and the cache still works correctly with the EC in the cache key.